### PR TITLE
Fix Tooltip on disabled button

### LINKED ISSUE #1816: Tooltip和disabled Button一起使用，mouseout的时候Tooltip没有隐藏
http://codepen.io/zation/pen/NNQjaY?editors=001

从button上mouseout不会隐藏tooltip，必须要hover到tooltip上，再mouseout，才会隐藏

### LINKED ISSUE #2097: Button禁用后Tooltip不会正常消失。
qq 20160618093113
截图如上，其他地方Tooltip的表现是离开该组件之后会自动消失，但是如果是在禁用后的Button上的话则不会消失了。测试代码如下：
`<Tooltip title="禁用后Tooltip">
          <Button disabled>禁用按钮</Button>
        </Tooltip>`
实际上一开始是用的`<Tooltip title="编辑"><Button type="ghost" shape="circle" icon="edit"></Tooltip>`这种方式，后来测试的时候写成上面那种格式。

### LINKED ISSUE #2381: Tooltip 和 Button(disabled) 混用BUG
#### 本地环境
- antd 版本: https://npmcdn.com/antd/dist/antd.js
- 操作系统及其版本：windows10
- 浏览器及其版本: chrome 51.0.2704.106

#### 你做了什么？

对 disabled 的 Button 使用 Tooltip 进行提示，提示不会隐藏

#### 你期待的结果是：

Tooltip 能正常使用，hover 显示，out 隐藏

#### 实际上的结果：

hover 显示，out 仍然显示

#### 可重现的在线演示

http://codepen.io/raygu/pen/akEzGP?editors=001

### LINKED ISSUE #4346: Tooltip acts weird on the disabled button inside the table row.
antd: "2.5.2",
windows 7 x64
chrome 55.0.2883.87

Hello. I am a big fan on antd!! but have a little problem recently.
I am trying to add tooltip around the buttons inside the table. In certain circumstance I need to have some buttons disabled. But tooltips keep showing even after the cursor moves away.
Also, even though i cannot replicate the error in the codepen demo, the tooltip location shifted a little bit. I record that in the screenshot. But this is not a big deal comparing to the previous one i guess.

Thanks

Demo: http://codepen.io/zj1926/pen/jVRvme?editors=001
screenshot: 
image

### LINKED ISSUE #4602: popover/tooltip chilren为button disabled 时 hoverout 失效
#### Environment(required)

- antd version:1.11.1

#### What happen?

popover/tooltip chilren为button disabled 时 hoverout 失效

#### Re-producible online demo
````html
<Tooltip title={"test"}
        placement={'right'}
 >
            <Button disabled type="primary" className="ant-tab-btn">test</Button>
</Tooltip>
````

### PR DESCRIPTION
**Pull request by afc163**
- ref https://github.com/react-component/tooltip/issues/18
- Close #1816
- Close #4346
- Close #2381
- Close #4346 
- Close #4602
- Close #2097

### CODE REVIEW

**Comment by afc163** [comment]
覆盖率为啥会下降？

**Comment by yesmeck** [comment]
@ @ 不是很清楚。。因为现在的覆盖率本来就是有点不准的，是按 Babel 转换后的代码计算的，而不是按 TypeScript 计算的。

**Comment by afc163** [comment]
变成 unknown 了...

File: components/tooltip/__tests__/tooltip.test.js [hunk 1]
```diff
@@ -49,4 +50,65 @@ describe('Tooltip', () => {
...
+        onVisibleChange={onVisibleChange}
+      >
+        <button disabled>Hello world!</button>
+      </Tooltip>
+    );
+
+    const button = wrapper.find('button').at(0);
+    button.simulate('mouseenter');
+    expect(onVisibleChange).not.toHaveBeenCalled();
+    expect(wrapper.ref('tooltip').prop('visible')).toBe(false);
```
Comment (benjycui) [inline components/tooltip/__tests__/tooltip.test.js]: According to me, `Tooltip`'s visible should be true and `onVisibleChange` should be called when mouse enter `button`, right?

File: components/tooltip/__tests__/tooltip.test.js [hunk 2]
```diff
@@ -49,4 +50,65 @@ describe('Tooltip', () => {
...
+
+    button.simulate('mouseleave');
+    expect(onVisibleChange).not.toHaveBeenCalled();
```
Comment (benjycui) [inline components/tooltip/__tests__/tooltip.test.js]: This should be called, too.

**Comment by JonySheep** [comment]
所以为什么会出现这个问题有结论么？看commit代码没看太明白，求解答(V) (°,,,,°) (V)

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/issues/21404#issuecomment-586800984

**Pull request merged**